tests/unit/sagemaker/workflow/test_pipeline.py

```diff
@@ -216,6 +216,129 @@ def test_pipeline_describe(sagemaker_session_mock):
+
+    sorted(output, key=lambda x: x["StepName"])
+    sorted(expected, key=lambda x: x["StepName"])
+
+    for i in range(len(output)):
+        assert output[i]["StepName"] == expected[i]["StepName"]
+        output_outBoundEdges = sorted(output[i]["OutBoundEdges"], key=lambda x: x["nextStepName"])
+        expected_outBoundEdges = sorted(
+            expected[i]["OutBoundEdges"], key=lambda x: x["nextStepName"]
+        )
+        assert len(output_outBoundEdges) == len(expected_outBoundEdges)
+        for j in range(len(output_outBoundEdges)):
+            assert output_outBoundEdges[j] == expected_outBoundEdges[j]
+
+
+def test_pipeline_build_adjacency_list_with_condition_edges_with_condition_step(
+    sagemaker_session_mock,
+):
+
+    ifStep1 = CustomStep("IfStep1")
+    ifStep2 = CustomStep("IfStep2")
+    elseStep1 = CustomStep("ElseStep1")
+    elseStep2 = CustomStep("ElseStep2")
+    normalStep1 = CustomStep(
+        "NormalStep", input_data=[elseStep2.properties.ModelArtifacts.S3ModelArtifacts]
+    )
+
+    conditionStep = ConditionStep(
+        name="ConditionStep",
+        conditions=[],
+        if_steps=[ifStep1, ifStep2],
+        else_steps=[elseStep1, elseStep2],
+    )
+
+    pipeline = Pipeline(
+        name="MyPipeline",
+        parameters=[],
+        steps=[conditionStep, normalStep1],
+        sagemaker_session=sagemaker_session_mock,
+    )
+
+    pipelineGraph = PipelineGraph.from_pipeline(pipeline)
+    output = pipelineGraph.build_adjacency_list_with_condition_edges()
+
+    expected = [
+        {
+            "StepName": "ConditionStep",
+            "OutBoundEdges": [
+                {"nextStepName": "ElseStep2", "edgeLabel": "False"},
+                {"nextStepName": "ElseStep1", "edgeLabel": "False"},
+                {"nextStepName": "IfStep2", "edgeLabel": "True"},
+                {"nextStepName": "IfStep1", "edgeLabel": "True"},
+            ],
+        },
+        {
+            "StepName": "ElseStep2",
+            "OutBoundEdges": [{"nextStepName": "NormalStep", "edgeLabel": None}],
+        },
+        {"StepName": "IfStep1", "OutBoundEdges": []},
+        {"StepName": "IfStep2", "OutBoundEdges": []},
+        {"StepName": "ElseStep1", "OutBoundEdges": []},
+        {"StepName": "NormalStep", "OutBoundEdges": []},
+    ]
+
+    assert len(output) == len(expected)
+
+    sorted(output, key=lambda x: x["StepName"])
+    sorted(expected, key=lambda x: x["StepName"])
+
+    for i in range(len(output)):
+        assert output[i]["StepName"] == expected[i]["StepName"]
+        output_outBoundEdges = sorted(output[i]["OutBoundEdges"], key=lambda x: x["nextStepName"])
+        expected_outBoundEdges = sorted(
+            expected[i]["OutBoundEdges"], key=lambda x: x["nextStepName"]
+        )
+        assert len(output_outBoundEdges) == len(expected_outBoundEdges)
+        for j in range(len(output_outBoundEdges)):
+            assert output_outBoundEdges[j] == expected_outBoundEdges[j]
+
```

Comment:
This seems to be duplicate code. Can we extract it to a helper function?

Reply:
Also, please add a test for StepCollection e.g. ModelStep in the if/else branch of a condition step.
example: https://github.com/aws/sagemaker-python-sdk/blob/9369a8781da9be92e1850ff62b42428cd61e23a6/tests/unit/sagemaker/workflow/test_model_step.py#L667-L677